src/core/ComplexFilter/index.tsx

```diff
@@ -18,23 +18,29 @@ export type ComplexFilterValue =
   | DefaultMenuSelectOption
   | null;
 
-interface ComplexFilterProps {
+export type Value<T, Multiple> = Multiple extends undefined | false
+  ? T | null
+  : Array<T> | null;
+
+interface ComplexFilterProps<Multiple> {
   label: string;
   options: DefaultMenuSelectOption[];
-  multiple?: boolean;
+  multiple?: Multiple;
   search?: boolean;
-  onChange: (options: ComplexFilterValue) => void;
+  onChange: (options: Value<DefaultMenuSelectOption, Multiple>) => void;
   MenuSelectProps?: Partial<typeof MenuSelect>;
   InputDropdownProps?: Partial<InputDropdownPropsType>;
-  value?: ComplexFilterValue;
+  value?: Value<DefaultMenuSelectOption, Multiple>;
   style?: React.CSSProperties;
   className?: string;
   PopperComponent?: typeof StyledPopper;
   PaperComponent?: typeof StyledPaper;
   InputDropdownComponent?: typeof InputDropdown;
 }
 
-export default function ComplexFilter({
+export default function ComplexFilter<
+  Multiple extends boolean | undefined = false
```

Comment:
... This part here, where the actual ComplexFilter component is defined. I don't understand how the Multiple filters through the definition and Type declaration and back to the underlying interface. Really curious to have you walk me through it.